Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -788,40 +788,36 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     if shouldSkip(node) {
       return .skipChildren
     }
-    if let unexpected = node.unexpectedBetweenDeinitKeywordAndEffectSpecifiers,
-      let name = unexpected.presentTokens(satisfying: { $0.tokenKind.isIdentifier == true }).only
-    {
-      addDiagnostic(
-        name,
-        .deinitCannotHaveName,
-        fixIts: [
-          FixIt(message: RemoveNodesFixIt(name), changes: .makeMissing(name))
-        ],
-        handledNodes: [name.id]
-      )
-    }
-    if let unexpected = node.unexpectedBetweenDeinitKeywordAndEffectSpecifiers,
-      let params = unexpected.compactMap({ $0.as(FunctionParameterClauseSyntax.self) }).only
-    {
-      addDiagnostic(
-        params,
-        .deinitCannotHaveParameters,
-        fixIts: [
-          FixIt(message: RemoveNodesFixIt(params), changes: .makeMissing(params))
-        ],
-        handledNodes: [params.id]
-      )
-    }
-    if let unexpected = node.unexpectedBetweenEffectSpecifiersAndBody,
-      let returnType = unexpected.compactMap({ $0.as(ReturnClauseSyntax.self) }).only
-    {
+
+    let name: TokenSyntax? = node
+      .unexpectedBetweenDeinitKeywordAndEffectSpecifiers?
+      .presentTokens(satisfying: \.tokenKind.isIdentifier)
+      .only
+
+    let params: FunctionParameterClauseSyntax? = node
+      .unexpectedBetweenDeinitKeywordAndEffectSpecifiers?
+      .compactMap({ $0.as(FunctionParameterClauseSyntax.self) })
+      .only
+
+    let returnType: ReturnClauseSyntax? = node
+      .unexpectedBetweenEffectSpecifiersAndBody?
+      .compactMap({ $0.as(ReturnClauseSyntax.self) })
+      .only
+
+    let nodes = [Syntax(name), Syntax(params), Syntax(returnType)].compactMap { $0 }
+
+    if !nodes.isEmpty {
       addDiagnostic(
-        returnType,
-        .deinitCannotHaveReturnType,
+        nodes.first!,
```

Comment:
Ah, one more thought that just crossed my mind.

```swift
    if let firstNode = nodes.first {
      addDiagnostic(
        returnType,
        .deinitCannotHaveReturnType,
        firstNode,
```

Allows us to get rid of the force-unwrap.